python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -370,21 +440,101 @@ def fit(self, X, y):
 
         return self
 
-    def predict(self, X):
+    def predict(self, X, deep_check=False):
         """
         Predicts the labels for X.
 
+        Parameters
+        ----------
+        X : dask cuDF dataframe or numpy array (n_samples, n_features)
+        deep_check : boolean (default = False)
+                     Set it to True if you want to run an extensive check of
+                     the concatenated treelite forest created using the
+                     forest information from all the workers.
+                     Required only while using predict for binary
+                     classification.
+        Returns
+        ----------
+        y: dask cuDF or numpy array of shape (n_samples, 1)
+
+        """
+        for w in self.workers:
+            if self.rfs[w].result().multi_class == 1:
+                self.multi_class = 1
+
+        if self.multi_class == 1:
+            preds = self.predict_using_cpu(X)
+
+        else:
+            preds = self.predict_using_fil(X, deep_check)
+
+        return preds
+
+    def predict_using_fil(self, X, deep_check):
+
+        concat_mod_bytes = self.concat_treelite_models(deep_check)
+
+        c = default_client()
+        key = uuid1()
+        gpu_futures = c.sync(extract_ddf_partitions, X)
+        worker_to_parts = OrderedDict()
+        for w, p in gpu_futures:
+            if w not in worker_to_parts:
+                worker_to_parts[w] = []
+            worker_to_parts[w].append(p)
+
+        workers = set(map(lambda x: x[0], gpu_futures))
+
+        comms = CommsContext(comms_p2p=True)
+        comms.init(workers=workers)
+
+        worker_info = comms.worker_info(comms.worker_addresses)
+
+        key = uuid1()
+        partsToSizes = [(worker_info[wf[0]]["r"], c.submit(
```

Comment:
This is an issue in other codes I’m working in at the moment, but this camel case named variable looks out of place since we use use underscores for variables 
```suggestion
        parts_to_sizes = [(worker_info[wf[0]]["r"], c.submit(
```